### ISSUE TITLE
[Performance] High idle CPU usage

### ISSUE THREAD

**Issue by wezm** [getinsomnia/insomnia#355]
## Overview
- `Insomnia Version:` 5.5.2
- `Operating System:`  Arch Linux
- `Summary: ` Insomnia uses 5–10% CPU when idle

## How To Reproduce

1. Open Insomnia
2. Look at CPU usage in `top`

## Expected Results

* When Insomnia is not performing a request it should use little to no CPU.

## Other Notes

Sample entries as seen in `top`:

```
  PID USER      PR  NI    VIRT    RES  %CPU %MEM     TIME+ S COMMAND                                                                                           
10904 wmoore    20   0 1292.8m 150.7m   8.6  1.0   6:16.08 S insomnia.bin                                                                                      
10866 wmoore    20   0 1479.8m  56.6m   3.3  0.4   1:41.18 S insomnia.bin                                                                                      
```

Machine is a Lenovo T470s with Intel(R) Core(TM) i7-7600U CPU @ 2.80GHz. Linux kernel version 
4.11.9-1.

**Comment by gschier** [comment]
Hey @wezm.

Here is what I'm seeing on my end:

- Signed in and syncing --> hovers around 0.1% and jumps to ~10% every 3-5 seconds
- Not signed in --> hovers between 0.1% and 0.6%

Also, Insomnia shows ~5% CPU usage for the first ~30 seconds after launch. I assume this is due to Chromium trying to run and re-optimize the source code.

**Comment by wezm** [comment]
Hmm I see a constant ~8% CPU usage right now on an instance on Insomnia that's been running since yesterday, not signed in.

On Wed, Jul 26, 2017, at 07:52 AM, Gregory Schier wrote:
> Hey @wezm[1].


> Here is what I'm seeing on my end:


>  * Signed in and syncing --> hovers around 0.1% and jumps to ~10%
>    every 3-5 seconds
>  * Not signed in --> hovers between 0.1% and 0.6%> Also, Insomnia shows ~5% CPU usage for the first ~30 seconds after launch. I assume this is due to Chromium trying to run and re-optimize the source code.> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub[2], or mute the thread[3]. > 


Links:

  1. https://github.com/wezm
  2. https://github.com/getinsomnia/insomnia/issues/355#issuecomment-317883719

**Comment by ahmetkurukose** [comment]
Hi,

Insomnia Version: 5.7.4
Operating System: Ubuntu Linux 17.04
Summary: Insomnia uses 5–10% CPU when idle

I think the problem is related with the automatic sync feature. When I disable the sync for the workspace,  idle cpu usage dropped to 0.7%.

**Comment by ahmetkurukose** [comment]
Strace output when using %10 CPU: (it says  Resource temporarily unavailable over and over)

recvmsg(26, 0x7ffd0f7d2830, 0)          = -1 EAGAIN (Resource temporarily unavailable)
poll([{fd=18, events=POLLIN}, {fd=19, events=POLLIN}, {fd=26, events=POLLIN}, {fd=34, events=POLLIN}], 4, 0) = 0 (Timeout)
recvmsg(26, 0x7ffd0f7d2860, 0)          = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(18, 0x7ffd0f7d2820, 0)          = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(26, 0x7ffd0f7d2830, 0)          = -1 EAGAIN (Resource temporarily unavailable)
poll([{fd=18, events=POLLIN}, {fd=19, events=POLLIN}, {fd=26, events=POLLIN}, {fd=34, events=POLLIN}], 4, 15) = 0 (Timeout)
recvmsg(26, 0x7ffd0f7d2860, 0)          = -1 EAGAIN (Resource temporarily unavailable)
write(35, "!", 1)                       = 1
write(60, "\0", 1)                      = 1
futex(0x52f97e8, FUTEX_WAKE_PRIVATE, 1) = 1
write(35, "!", 1)                       = 1
recvmsg(18, 0x7ffd0f7d2820, 0)          = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(26, 0x7ffd0f7d2830, 0)          = -1 EAGAIN (Resource temporarily unavailable)
poll([{fd=18, events=POLLIN}, {fd=19, events=POLLIN}, {fd=26, events=POLLIN}, {fd=34, events=POLLIN}], 4, 0) = 1 ([{fd=34, revents=POLLIN}])
recvmsg(26, 0x7ffd0f7d2860, 0)          = -1 EAGAIN (Resource temporarily unavailable)
read(34, "!!", 2)                       = 2
recvmsg(18, 0x7ffd0f7d2820, 0)          = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(26, 0x7ffd0f7d2830, 0)          = -1 EAGAIN (Resource temporarily unavailable)
poll([{fd=18, events=POLLIN}, {fd=19, events=POLLIN}, {fd=26, events=POLLIN}, {fd=34, events=POLLIN}], 4, 0) = 0 (Timeout)
recvmsg(26, 0x7ffd0f7d2860, 0)          = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(18, 0x7ffd0f7d2820, 0)          = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(26, 0x7ffd0f7d2830, 0)          = -1 EAGAIN (Resource temporarily unavailable)
poll([{fd=18, events=POLLIN}, {fd=19, events=POLLIN}, {fd=26, events=POLLIN}, {fd=34, events=POLLIN}], 4, 15) = 0 (Timeout)
recvmsg(26, 0x7ffd0f7d2860, 0)          = -1 EAGAIN (Resource temporarily unavailable)
write(35, "!", 1)                       = 1
write(60, "\0", 1)                      = 1
futex(0x52f97e8, FUTEX_WAKE_PRIVATE, 1) = 1
write(35, "!", 1)                       = 1
recvmsg(18, 0x7ffd0f7d2820, 0)          = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(26, 0x7ffd0f7d2830, 0)          = -1 EAGAIN (Resource temporarily unavailable)
poll([{fd=18, events=POLLIN}, {fd=19, events=POLLIN}, {fd=26, events=POLLIN}, {fd=34, events=POLLIN}], 4, 0) = 1 ([{fd=34, revents=POLLIN}])
recvmsg(26, 0x7ffd0f7d2860, 0)          = -1 EAGAIN (Resource temporarily unavailable)
read(34, "!!", 2)                       = 2
recvmsg(18, 0x7ffd0f7d2820, 0)          = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(26, 0x7ffd0f7d2830, 0)          = -1 EAGAIN (Resource temporarily unavailable)
poll([{fd=18, events=POLLIN}, {fd=19, events=POLLIN}, {fd=26, events=POLLIN}, {fd=34, events=POLLIN}], 4, 0) = 0 (Timeout)
recvmsg(26, 0x7ffd0f7d2860, 0)          = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(18, 0x7ffd0f7d2820, 0)          = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(26, 0x7ffd0f7d2830, 0)          = -1 EAGAIN (Resource temporarily unavailable)
poll([{fd=18, events=POLLIN}, {fd=19, events=POLLIN}, {fd=26, events=POLLIN}, {fd=34, events=POLLIN}], 4, 16^Cstrace: Process 13948 detached


https://stackoverflow.com/questions/17666145/why-strace-shows-eagain-resource-temporarily-unavailable

**Comment by gschier** [comment]
@ahmetkurukose ya, the sync feature could probably use some optimization. 

I also recently noticed that, if any part of the Insomnia app is visible, CPU idles a lot higher than when none of it is. This makes me feel like there's probably some bit of code triggering a redraw of the app.

**Comment by gschier** [comment]
Just pushed up a few performance fixes in #411 

Will try to get to some of the less-easy ones soon.
